vpn/tls.go

```diff
@@ -39,9 +42,10 @@ func initTLS(session *session, opt *Options) (*tls.Config, error) {
 		// VerifyConnection or VerifyPeercertificate callback?
 		// ServerName:         "vpnserver",
 		// VerifyPeerCertificate: ...,
-		InsecureSkipVerify: true,
-		MinVersion:         tls.VersionTLS12,
-		MaxVersion:         uint16(max),
+		InsecureSkipVerify:          true,
+		MinVersion:                  tls.VersionTLS12,
+		MaxVersion:                  uint16(max),
```

Comment:
IIUC, it's irrelevant to set these fields given we're using parroting; is that so?

Reply:
I see this comment around the line you linked to:

```
// if version is not set explicitly in the ClientHelloSpec, check the SupportedVersions extension
```

do we know whether the client hello spec we're using enforces a version already? Should we trust the client hello spec to provide us with the right min/max for better parroting?

Reply:
so, this is interesting. For compatibility with a particular provider, I was using a very particular client hello with only 5 ciphersuites. I had passed a `--tls-version-min=1.2` to openvpn, so in the end the client hello spec included the `supported_versions` extension with `1.2` and `1.3`.

If I capture a reference hello spec setting `--tls-version-min=1.2` and `--tls-version-max=1.2`, then the `supported_versions` extension is omitted from the spec, and the `version: TLS 1.2` header prevails. 

Now, if I don't pass any `--tls-cipher` in the call to openvpn to capture a spec, I get 22 ciphersuites for `tls=1.2`, and 50 ciphersuites for `tls-max=1.3`.

OTOH, I don't need to pin the old discrete DH extension anymore (that particular problem has been solved by other means), so what I think I would do is:

1. to remove any capability to set min or max tls version in the TLS handshake, and document that we're parroting tls 1.2 and 1.3. with a broad, by default ciphersuite selection (this would leave all the functions to parse tls version ineffectual).
2. deal with any incompatibilities coming from specific ciphersuites in the future (probably adapting the reference client hello).

Reply:
for completeness:

> do we know whether the client hello spec we're using enforces a version already?

from this capture (see my explanation above) I _think_ openvpn is always setting explicitely the `TLS Version` header to the min, and then if the `--tls-max-version` flag differs from the `min`, it adds the `SupportedVersions` extension.

![tls-1 3-version](https://user-images.githubusercontent.com/99027643/173813152-b8be520d-4630-4074-b64b-11b928f4ae28.png)

Reply:
[...]
> OTOH, I don't need to pin the old discrete DH suite anymore (that particular problem has been solved by other means), so what I think I would do is:
> 
>     1. to remove any capability to set min or max tls version in the TLS handshake, and document that we're parroting tls 1.2 and 1.3. with a broad, by default ciphersuite selection (this would leave all the code to parse tls version from the config file ineffectual. I am tempted to delete those too, and bring it back if we ever need it in the future).
> 
>     2. deal with any incompatibilities coming from specific ciphersuites in the future (probably adapting the reference client hello).

Sounds good! You should apply these changes and merge! (I am going to green light the PR in the meanwhile!)

Thank you!

🐳